src/parser.rs

```diff
@@ -15,90 +65,97 @@ fn is_digit(c: char) -> bool {
     }
 }
 
-named!(letter<CompleteStr, char>,
+named!(letter<Span, char>,
     one_of!("ABCDEFGHIJKLMNOPQRSTUVWXYZ"));
 
-named!(digit<CompleteStr, u8>,
+named!(digit<Span, u8>,
     map!(one_of!("0123456789"), |c| c as u8 - b'0'));
 
 // 5. Programs
 
-named!(pub program<CompleteStr, Result<Program, Error>>,
+named!(pub program<Span, Result<Program, Error>>,
     do_parse!(
         blocks: many0!(terminated!(block, end_of_line)) >>
         opt!(end_of_line) >>
         (Program::new(blocks))
     ));
 
-named!(pub block<CompleteStr, Block>,
+named!(block<Span, Block>,
     do_parse!(
         line_number: line_number >>
         statement: sep!(space, statement) >>
         space0 >>
-        (Block::Line{ line_number, statement })
+        (Block::Line{
+            line_number,
+            statement: statement.0,
+            statement_source: statement.1
```

Comment:
What is `statement_source`? The original source string?

Reply:
If you take a look at definition of `Block`, then you will see this is a `Span` which references the source code of the statement.